### ISSUE TITLE
Wrong diff ID calculated error when building container image

### ISSUE THREAD

**Issue by alexbrand** [vito/oci-build-task#22]
I am getting the following error when trying to build an image:

```
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0:
 failed to build LLB: wrong diff id calculated on extraction "sha256:d5d063acba121b99927d52088ad7dfd9ee982880626fb10e1aef5f3ead99892d"
```

This is the pipeline configuration:
```
resources:
  - name: code-repo
    type: git
    source:
      branch: master
      uri: https://github.com/alexbrand/spring-boot-rest-example.git

  - name: container-image
    type: registry-image
    source:
      repository: ((docker-hub-account))/spring-boot-rest-example
      username: ((docker-hub-user))
      password: ((docker-hub-pass))

jobs:
  - name: build-push-image
    plan:
    - get: code-repo
      trigger: true

    - task: build
      privileged: true
      config:
        platform: linux
        image_resource:
          type: registry-image
          source:
            repository: vito/oci-build-task
        inputs:
        - name: code-repo
          path: .
        outputs:
        - name: image
        run:
          path: build

    - put: container-image
      params:
        additional_tags: code-repo/.git/short_ref
        image: image/image.tar
```

Full build logs:

```
fetching vito/oci-build-task@sha256:cfb2983956145f54a4996c2aff5fc598856c8722922a6e73f9ebfa3d9b3f9813
4fe2ade4980c [==========================================] 2.1MiB/2.1MiB
23e16c159a3e [==========================================] 7.5MiB/7.5MiB
c455d41ca53a [==============================================] 899b/899b
65f69dce8edb [========================================] 22.9MiB/22.9MiB
3df4123dfd16 [==========================================] 2.1MiB/2.1MiB
c266d1ba4b08 [==========================================] 1.7MiB/1.7MiB
6a3b4b7d898b [==============================================] 744b/744b
#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 859B done
#2 DONE 0.0s

#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#4 [internal] load metadata for docker.io/library/maven:3.6.3-jdk-11-slim
#4 ...

#3 [internal] load metadata for gcr.io/distroless/java:11
#3 DONE 0.6s

#4 [internal] load metadata for docker.io/library/maven:3.6.3-jdk-11-slim
#4 DONE 0.7s

#8 [internal] load build context
#8 transferring context: 39.90kB done
#8 DONE 0.0s

#6 [maven_tool_chain_cache 1/8] FROM docker.io/library/maven:3.6.3-jdk-11-s...
#6 resolve docker.io/library/maven:3.6.3-jdk-11-slim@sha256:f53c375e8391df1b005aaa5d59145d457e8cf3fd2ba4b562baeef44a471323fa done
#6 sha256:bc51dd8edc1b1132bb97827ed6bd16aac332b03fb03d4c02d2088067a5fbb499 56.44kB / 27.09MB 0.1s
#6 sha256:186d54f60a04eacc9acb9f6fdede650475e3c231ec231635607b7df11ec34063 0B / 9.58MB 0.1s
#6 sha256:27378801d6060c82aa6e8c5bc80672354f80a9a3a50371fa790876d5258adb1f 360B / 360B 0.1s done
#6 sha256:adf4d10a782b71ea3c226ce20c1acdfb0aeefb137fb5c5fa3100aaa275ec71cd 210B / 210B 0.1s done
#6 sha256:2a41cd9e737b0a2b6625263325f462a8991a0e66eaa35dc73ef9c26f58dd6608 9.49kB / 9.49kB done
#6 sha256:d1d06863bb82bdc3f4cd17d49c0dee8720f91c4d9e40cce1b4a691f9459266be 0B / 3.25MB 0.1s
#6 sha256:7c35c75d6bdb8ca6f0ab362d68d1960320b4d16c43386b76993970907198bfde 0B / 2.61MB 0.1s
#6 sha256:5c2041c657279f379fd8b1ad2e608910966ae0c123d3eeeb1ffd8d0429624fba 0B / 196.34MB 0.1s
#6 sha256:f53c375e8391df1b005aaa5d59145d457e8cf3fd2ba4b562baeef44a471323fa 549B / 549B done
#6 sha256:82e384dd71920dfb1b098549272bf3a348bd242d1afdd0baff1507ced69c77a2 2.00kB / 2.00kB done
#6 sha256:50115f90b29477cffbad6f7c4e384bc901d9a9b24d511f8b60ed8de4b9ecb6c2 0B / 856B 0.1s
#6 sha256:bc51dd8edc1b1132bb97827ed6bd16aac332b03fb03d4c02d2088067a5fbb499 3.24MB / 27.09MB 0.3s
#6 sha256:186d54f60a04eacc9acb9f6fdede650475e3c231ec231635607b7df11ec34063 4.03MB / 9.58MB 0.3s
#6 sha256:d1d06863bb82bdc3f4cd17d49c0dee8720f91c4d9e40cce1b4a691f9459266be 2.78MB / 3.25MB 0.3s
#6 sha256:7c35c75d6bdb8ca6f0ab362d68d1960320b4d16c43386b76993970907198bfde 2.61MB / 2.61MB 0.3s done
#6 sha256:50115f90b29477cffbad6f7c4e384bc901d9a9b24d511f8b60ed8de4b9ecb6c2 856B / 856B 0.1s done
#6 sha256:bc51dd8edc1b1132bb97827ed6bd16aac332b03fb03d4c02d2088067a5fbb499 9.49MB / 27.09MB 0.6s
#6 sha256:186d54f60a04eacc9acb9f6fdede650475e3c231ec231635607b7df11ec34063 9.58MB / 9.58MB 0.5s done
#6 sha256:d1d06863bb82bdc3f4cd17d49c0dee8720f91c4d9e40cce1b4a691f9459266be 3.25MB / 3.25MB 0.3s done
#6 sha256:5c2041c657279f379fd8b1ad2e608910966ae0c123d3eeeb1ffd8d0429624fba 17.33MB / 196.34MB 0.6s
#6 sha256:bc51dd8edc1b1132bb97827ed6bd16aac332b03fb03d4c02d2088067a5fbb499 22.08MB / 27.09MB 0.9s
#6 sha256:5c2041c657279f379fd8b1ad2e608910966ae0c123d3eeeb1ffd8d0429624fba 31.66MB / 196.34MB 0.9s
#6 sha256:bc51dd8edc1b1132bb97827ed6bd16aac332b03fb03d4c02d2088067a5fbb499 27.09MB / 27.09MB 1.0s
#6 sha256:bc51dd8edc1b1132bb97827ed6bd16aac332b03fb03d4c02d2088067a5fbb499 27.09MB / 27.09MB 1.1s done
#6 sha256:5c2041c657279f379fd8b1ad2e608910966ae0c123d3eeeb1ffd8d0429624fba 65.92MB / 196.34MB 1.3s
#6 sha256:5c2041c657279f379fd8b1ad2e608910966ae0c123d3eeeb1ffd8d0429624fba 83.71MB / 196.34MB 1.5s
#6 ...

#5 [stage-1 1/2] FROM gcr.io/distroless/java:11@sha256:b715126ebd36e5d5c2fd...
#5 resolve gcr.io/distroless/java:11@sha256:b715126ebd36e5d5c2fd730f46a5b3c3b760e82dc18dffff7f5498d0151137c9 done
#5 sha256:b715126ebd36e5d5c2fd730f46a5b3c3b760e82dc18dffff7f5498d0151137c9 1.16kB / 1.16kB done
#5 sha256:c5806e11a467bea9d9206123cee9e4c1c9fae3ceed1629abfe080166de318f38 63.52MB / 63.52MB 1.4s done
#5 sha256:24f0c933cbef83faee52f82c7f889c727b1ece5123b92d036c52fa865480f037 657.29kB / 657.29kB 0.2s done
#5 sha256:37c25e886b24e7c9999e53f310aa87066a2cc8fabba0e866e4769854f0f2803a 965B / 965B done
#5 sha256:69e2f037cdb30c8d329b17dad42cd9d92a45d93c17e6699650b23c55eceacb5f 7.33MB / 7.33MB 0.4s done
#5 sha256:3e010093287c245d72a774033b4cddd6451a820bfbb1948c97798e1838858dd2 643.66kB / 643.66kB 0.2s done
#5 unpacking gcr.io/distroless/java:11@sha256:b715126ebd36e5d5c2fd730f46a5b3c3b760e82dc18dffff7f5498d0151137c9 0.2s done
#5 ERROR: wrong diff id calculated on extraction "sha256:d5d063acba121b99927d52088ad7dfd9ee982880626fb10e1aef5f3ead99892d"

#6 [maven_tool_chain_cache 1/8] FROM docker.io/library/maven:3.6.3-jdk-11-s...
#6 CANCELED
------
 > [stage-1 1/2] FROM gcr.io/distroless/java:11@sha256:b715126ebd36e5d5c2fd730f46a5b3c3b760e82dc18dffff7f5498d0151137c9:
------
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to build LLB: wrong diff id calculated on extraction "sha256:d5d063acba121b99927d52088ad7dfd9ee982880626fb10e1aef5f3ead99892d"
FATA[0006] failed to build: build: exit status 1        
FATA[0006] failed to run task: exit status 1  
```

**Comment by cirocosta** [comment]
seems like it relates to https://github.com/containerd/containerd/pull/3528

I don't think we're using `containerd` as buildkit's backend in `oci-build-task`, but from what I can tell taking a quick glance at [`pull.go`](https://github.com/moby/buildkit/blob/89978e706966b5662f2d108c906084d644559178/util/pull/pull.go#L9) from buildkit, it seems like it does interact with the piece of code that was patched in that issue I mentioned

might be worth bumping it?

**Comment by alexbrand** [comment]
Thanks, @cirocosta. Do you mind clarifying what you mean by "might be worth bumping it?"

**Comment by alexbrand** [comment]
For what it's worth, creating a fresh image of the Resource (`vito/oci-build-task`), which in turn gets a fresh image of `moby/buildkit:master` got me past this. Thanks @cirocosta for the pointer.

**Comment by cirocosta** [comment]
sorry for the delay @alexbrand , but cooool to see that it worked! :grin: 

@vito sounds like a good time to push a new image :grin:  btw, do you see it being versioned any time soon?

**Comment by pnsantos** [comment]
Just bumped into this... @vito any chance a fresh build could be run and the image pushed to dockerhub?

**Comment by cirocosta** [comment]
(vito is on vacations at the moment, back soon 😁)

**Comment by vito** [comment]
this is fixed in 0.2.0 - sorry for the wait!
